src/components/apply/formValidators.js

```diff
@@ -1,27 +1,62 @@
-validations = {
-  firstName: [required()],
-  lastName: [required()],
-  middleInitial:[required()],
-  soc: [required(), length({is: 11})],
-  phoneNumber: [required(), length({is: 14})],
-  phoneNumberType: [required()],
-  email:    [required(), email()],
-  dob:      [
-    required(),
-    numericality({ int: true }),
-    numericality({ '>=': 18, msg: 'Must be 18 years or older to apply.' })
-  ],
+import React from 'react';
+import _ from 'lodash';
 
+import { FormattedMessage } from 'react-intl';
+import { required, email, addValidator } from 'redux-form-validators';
 
-};
+const validationMsg = (message) => (
+  <FormattedMessage id='form.error' defaultMessage={message} />
+);
 
-validate = (values) => {
-  const errors = {}
-  for (let field in validations) {
-    let value = values[field]
-    errors[field] = validations[field].map(validateField => {
-      return validateField(value, values)
-    }).find(x => x)
+const dobValidation = addValidator({
+  defaultMessage: validationMsg('Must be 18 years or older.'),
+  validator: (options, value) => {
+    let ageMs = Date.now() - new Date(value).getTime();
+    return (new Date(ageMs).getUTCFullYear() - 1970) >= 18;
   }
-  return errors;
-}
+});
+
+const stripFormatting = input => _.replace(input, /[^0-9]+/g, '');
+
+const numberValidationWithLength = addValidator({
+  defaultMessage: validationMsg('Invalid Input'),
+  validator: (options, value, allValues) =>{
+    return stripFormatting(value).length === options.length;
+  }
+});
+
+const requiredInput = required({msg: validationMsg('Required')});
+const validations = {
+  firstName: [requiredInput],
+  lastName:[requiredInput],
+  soc:[requiredInput, numberValidationWithLength({length: 9})],
+  dob:[requiredInput, dobValidation()],
+  phoneNumber: [requiredInput, numberValidationWithLength({length: 10})],
+  phoneNumberType:[requiredInput],
+  emailAddress:[requiredInput, email({msg: validationMsg('Invalid Email')})],
+  address:[requiredInput],
+  city: [requiredInput],
+  state:[requiredInput],
+  zipCode:[requiredInput, numberValidationWithLength({length: 5})]
+};
+
+// Applies all the validator functions to the input, returning the value of the first rejection
+// or `undefined`.
+const applyValidators = (validators, value, values) => _.find(_.map(validators, validator => validator(value, values)));
+
+/* This function is adapted from code from redux-form-validators
```

Comment:
That's weird that it isn't built in. Definitely prefer the version you have now to the one posted. Might be worthwhile to `_.mapValues` and see if you can make an even simpler version.